mutation_reader.cc

```diff
@@ -646,7 +660,7 @@ flat_mutation_reader make_combined_reader(schema_ptr schema,
     if (readers.size() == 1) {
         return std::move(readers.front());
     }
-    return make_flat_mutation_reader<combined_mutation_reader>(schema,
+    return make_combined_reader(std::move(schema),
```

Comment:
this move relies on UB (https://en.cppreference.com/w/cpp/language/eval_order), please get rid of it

Reply:
Good catch, fixed.